src/app/shared/search/search-result.html

```diff
@@ -17,11 +16,18 @@
   </thead>
   <tbody>
   <tr *ngFor="let result of resultView.results | ccdSortSearchResult : sortParameters | paginate: { itemsPerPage: paginationPageSize, currentPage: selected.page, totalItems: paginationMetadata.total_results_count }">
-    <td class="caseid-col">
-      <a routerLink="/case/{{jurisdiction.id}}/{{caseType.id}}/{{result.case_id}}"><div class="text-16" [style.visibility]="hideRows ? 'hidden' : 'visible'">{{result.case_id | ccdCaseReference}}</div></a>
-    </td>
-    <td *ngFor="let col of resultView.columns">
-      <div class="text-16" [style.visibility]="hideRows ? 'hidden' : 'visible'">
+    <td *ngFor="let col of resultView.columns; let colIndex = index">
+      <a *ngIf="colIndex == 0" routerLink="/case/{{jurisdiction.id}}/{{caseType.id}}/{{result.case_id}}">
+        <div class="text-16" [style.visibility]="hideRows ? 'hidden' : 'visible'">
+          <ccd-field-read [caseField]="{
+                  id: col.case_field_id,
+                  label: col.label,
+                  field_type: col.case_field_type,
+                  value: result.case_fields[col.case_field_id]? col.case_field_id === 'case_reference' ? (result.case_fields[col.case_field_id] | ccdCaseReference) : result.case_fields[col.case_field_id] : (result.case_id | ccdCaseReference)
```

Comment:
I fear this won't work if the first column's field type is not Text or Number (e.g. fixed list, document, moneyGBP, ...).

I would recommend replacing the ternary conditions on `ccd-field-read`'s value, by an if/else block with different templates as this would resolve the above concern.

Reply:
Thanks. I have corrected the code as discussed using templates. Please re-review.